az-cvm-vtpm/src/vtpm/mod.rs

```diff
@@ -88,16 +92,80 @@ pub enum ReportError {
 
 /// Get a HCL report from an nvindex
 pub fn get_report() -> Result<Vec<u8>, ReportError> {
+    let (nv_index, mut context) = get_session_context()?;
+
+    let report = nv::read_full(&mut context, NvAuth::Owner, nv_index)?;
+    Ok(report)
+}
+
+/// Get a HCL report from an nvindex using temporary nvindex and report_data which will trigger
+/// refresh on main nvindex. Given report_data will be stored in HCL report in user_data field
+pub fn get_report_with_data(report_data: &[u8]) -> Result<Vec<u8>, ReportError> {
+    let (nv_index, mut context) = get_session_context()?;
+
+    let nv_index_temp = NvIndexTpmHandle::new(VTPM_HCL_REPORT_NV_INDEX_TEMP)?;
+    write_nv_index(&mut context, nv_index_temp, report_data);
+
+    // Adding a sleep for 3s for the user data to be reflected in VTPM_HCL_REPORT_NV_INDEX
+    thread::sleep(core::time::Duration::from_secs(3));
```

Comment:
Additional info: there was a bug from Microsoft end due to which it was retrieving stale quote. It was added as a workaround to get past the stale quote issue. Issue was solved so this will be removed